discovery/src/main/scala/com.wavesplatform.discovery/collections/ExpirationSet.scala

```diff
@@ -0,0 +1,26 @@
+package com.wavesplatform.discovery.collections
+
+class ExpirationSet[T](expirationTimeMilis: Long) extends scala.collection.mutable.Set[T]{
```

Comment:
you can use guava's `cache` like in `MicroblockSyncronizer`: 
`  private val nextInvs = cache[MicroBlockSignature, MicroBlockInv](settings.invCacheTimeout)`